Sources/Rendering/WebGPU/PixelSpaceCallbackMapper/index.js

```diff
@@ -0,0 +1,74 @@
+import macro from 'vtk.js/Sources/macro';
+import vtkViewNode from 'vtk.js/Sources/Rendering/SceneGraph/ViewNode';
+
+// ----------------------------------------------------------------------------
+// vtkWebGPUPixelSpaceCallbackMapper methods
+// ----------------------------------------------------------------------------
+
+function vtkWebGPUPixelSpaceCallbackMapper(publicAPI, model) {
+  model.classHierarchy.push('vtkWebGPUPixelSpaceCallbackMapper');
+
+  publicAPI.opaquePass = (prepass, renderPass) => {
+    model.WebGPURenderer = publicAPI.getFirstAncestorOfType(
+      'vtkWebGPURenderer'
+    );
+    model.WebGPURenderWindow = model.WebGPURenderer.getParent();
+    const aspectRatio = model.WebGPURenderer.getAspectRatio();
+    const camera = model.WebGPURenderer
+      ? model.WebGPURenderer.getRenderable().getActiveCamera()
+      : null;
+    const tsize = model.WebGPURenderer.getTiledSizeAndOrigin();
+    const texels = null;
+
+    if (model.renderable.getUseZValues()) {
+      // Todo
+    }
+
+    model.renderable.invokeCallback(
+      model.renderable.getInputData(),
+      camera,
+      aspectRatio,
+      tsize,
+      texels
+    );
+  };
+
+  publicAPI.queryPass = (prepass, renderPass) => {
+    if (prepass) {
+      if (model.renderable.getUseZValues()) {
+        renderPass.setDepthRequested(true);
+      } else {
+        renderPass.setDepthRequested(true);
+      }
```

Comment:
Why does the if/else call the same method?

Should this be something like `renderPass.setDepthRequested(model.renderable.getUseZValues())` with maybe a `!!` infront to make sure it is a boolean?

Reply:
Because it is a straight copy of the OpenGL one :-) Maybe @scottwittenburg knows but likely a bug.

Reply:
I think I know the correct logic and will give it a try

Reply:
The wording on the renderPass method is not great. It should be renderPass.requestDepth(); as you only call it when you want depth values, otherwise you leave it alone. For WebGPU it will probably be removed when I get to code that uses it as with WebGPU I always use a texture backed depth buffer.

Reply:
Actually I'm going to change that now as it isn't quite right.

Reply:
Sorry I don't recall, but it looks wrong obviously.  I'm guessing if we're not using z values, then it was supposed to be depth requested = false. 🤷

Reply:
Thanks @scottwittenburg   I think I have it fixed now, fingers crossed.